frontend/public/co-fetch.js

```diff
@@ -124,8 +124,8 @@ export const coFetchUtils = {
   parseJson,
 };
 
-export const coFetchJSON = (url, method = 'GET', options = {}) => {
-  const headers = { Accept: 'application/json' };
+export const coFetchCommon = (url, method = 'GET', options = {}, customHeaders = {}) => {
```

Comment:
Maybe just `coFetch` as the name?

Reply:
`coFetch` already exists in this file and is used in this method. Welcome to entertain another name, this was the best one I could come up with lol -- naming is hard 😄